### PR TITLE
ROX-21530: Updating Cert Expiry Metrics

### PR DESCRIPTION
**Pull request by aaa5kameric**
## Description
Background: Implement monitoring for certificate expiration, tracking and managing of digital certificates expiration dates. Extracts timestamps from certificates and exposes metrics to Prometheus. (FIXED)

In response to error in: https://github.com/stackrox/rhacs-observability-resources/pull/279
Problem: `acs_fleetshard_certificate_expiration_timestamp` - seems to retain all certificates even after their namespaces have been deleted. Combined with the ephemeral instances created by the probe, this creates a very large number of time series. 
Solution: Fixed the handling of metric deletion by:
-adding namespace informer
-implemented `DeletePartialMatch` instead of `Delete`

Previous Certificate Monitoring PR: https://github.com/stackrox/acs-fleet-manager/pull/1946
Previous Certificate Alerting PR: https://github.com/stackrox/rhacs-observability-resources/pull/276

## Tests
Created k9s environment locally with pods/namespaces. Used `create-central.sh` and `delete-central.sh` scripts to test local host metric creation and deletion.

### CODE REVIEW

File: internal/certmonitor/certmonitor_test.go [hunk 1]
```diff
@@ -333,6 +333,10 @@ func TestCertMonitor(t *testing.T) {
 		Data:       map[string][]byte{"tls-1.crt": generateCertWithExpiration(t, newExpiryTime)},
 	}
 
+	mockNamespace := &v1.Namespace{ // pragma: allowlist secret
+		ObjectMeta: metav1.ObjectMeta{Namespace: "namespace-1", Name: "secret-1"},
+	}
```
Comment (ludydoo) [inline internal/certmonitor/certmonitor_test.go:338]:
```suggestion
	mockNamespace := &v1.Namespace{ // pragma: allowlist secret
		ObjectMeta: metav1.ObjectMeta{Name: "namespace-1"},
	}
```

**Comment by kurlov** [comment]
You need to rebase to fix CI pre commit

File: internal/certmonitor/certmonitor_test.go [hunk 2]
```diff
@@ -291,7 +291,7 @@ func TestCertMonitor_secretMatches(t *testing.T) {
 }
 
 // TestCertMonitor func tests certificates event handlers + prometheus metrics handling
```
Comment (ludydoo) [inline internal/certmonitor/certmonitor_test.go:293]:
```suggestion
// TestCertMonitor_Secret func tests certificates event handlers + prometheus metrics handling
```

**Review by ludydoo — APPROVED** [review]

**Pull request merged**